build.sbt

```diff
@@ -652,9 +656,9 @@ lazy val bench = project
   .enablePlugins(JmhPlugin)
 
 lazy val binCompatTest = project
-  .disablePlugins(CoursierPlugin)
   .settings(noPublishSettings)
   .settings(
+    useCoursier := false,
```

Comment:
@larsrh Is this a workaround for the problem described in [#3026](https://github.com/typelevel/cats/pull/3026/commits/4ee38b22a2455c0a812b9bc86ef70e835744fcaa)? If yes, we could just bump sbt and deal with the MiMa update separately.

Reply:
Ah, I somehow didn't see that PR. But I can try to incorporate it here.

Reply:
That isn't necessary. The only change I did there that is not in this PR, is a change we don't want to merge. :-) I just wanted to note that we could bump sbt separately with the `useCoursier := false` workaround.

Reply:
Wait, now I'm confused. Is this PR good to merge or should I change anything?

Reply:
Oh, btw, we can't bump sbt without also bumping MiMa (there's some binary breakage there).

Reply:
Interesting, I didn't notice this in #3026.

Reply:
Try bumping to 1.3.2. At least I thought that's what happened in another PR, but maybe I was mistaken :thinking:

Reply:
You're right. Bumping from 1.3.0 to 1.3.2 produced errors like this in the "binary compatibility" job:
```
(coreJVM / mimaReportBinaryIssues) java.lang.NoSuchMethodError: com.typesafe.tools.mima.core.Settings.allSettings()Lscala/collection/mutable/HashSet;
```